modules/openapi-generator/src/main/resources/typescript/api/api.mustache

```diff
@@ -28,6 +29,11 @@ import { {{classname}} } from '..{{filename}}{{extensionForDeno}}';
 @injectable()
 {{/useInversify}}
 export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
+    private defaultSecurityAuthentication: SecurityAuthentication | undefined;
```

Comment:
What is the rationale of putting this as a separate object in the `RequestFactory` instead of putting it in `this.configuration`?

Reply:
this was done duplicating the old typescript generator's interface, I'm happy to switch it into `this.configuration`.
I can add it to the `Configuration` object as a `default` field with `SecurityAuthentication` type and then it could be overrode by the config passed in method calls as well